pkg/plugins/backendplugin/grpcplugin/client_v2.go

```diff
@@ -186,7 +195,7 @@ func (c *ClientV2) QueryData(ctx context.Context, req *backend.QueryDataRequest)
 			return nil, plugins.ErrMethodNotImplemented
 		}
 
-		return nil, fmt.Errorf("%v: %w", "Failed to query data", err)
+		return nil, handleGrpcStatusError(ctx, err)
```

Comment:
This is for `QueryData`. Do we also need it for the other methods `CheckHealth`, `CallResource`, etc?

Reply:
Yeah I think this is really only relevant in the context of query data right now since that's how SLIs/SLOs are measured. I'll let you weigh in though @ivanahuckova

Reply:
> I think this is really only relevant in the context of query data right now since that's how SLIs/SLOs are measured. 

+1, this is now needed only from QueryData.